Libraries/LibJS/Parser.cpp

```diff
@@ -1121,6 +1139,9 @@ void Parser::consume_or_insert_semicolon()
     // ...token is a closing curly brace
     if (match(TokenType::CurlyClose))
         return;
+    // ...token is "in" or "of"
+    if (match(TokenType::In) || match(TokenType::Of))
+        return;
```

Comment:
Is this mentioned somewhere in the ES spec? This function is supposed to adhere strictly to the semicolon insertion rules.

Reply:
I doubt it, this was more of a workaround to re-use the existing variable declaration parsing (`parse_variable_declaration()`) which tries to `consume_or_insert_semicolon()`, and the is none... But you're right, I need to think of a better solution.